client/components/exercises-desktop/exercise-accordion-desktop.jsx

```diff
@@ -32,10 +32,8 @@ export default class ExerciseAccordion extends React.Component {
     const body = { userId: 1, exerciseId, username: 'GIGACHAD', muscleId: this.props.muscleID, muscleName: this.props.muscleName };
     fetch('/api/saved-exercises', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(body) })
       .then(res => res.json())
-      .then(data => {
-      })
+      .then(this.checkSavedIcons())
```

Comment:
`.then()` takes a function definition as its argument, but you are passing the return value of an (immediate) function _call_.